### ISSUE TITLE
No information of lastlogin displayed

### ISSUE THREAD

**Issue by 74cmonty** [dsoares/Roundcube-Plugin-Lastlogin#13]
Hi!
I'm facing the issue that after successfully installing the plugins "lastlogin" and "geolocation" and modifying config-file <roundcube-dir>/plugins/lastlogin/config.inc.php.dist I cannot see any entries in the table although the plugin is activated correctly.
![2016-07-18_10-36-18](https://cloud.githubusercontent.com/assets/12523591/16909508/862a3de6-4cd3-11e6-8bb0-26be52ad3b59.png)

There's no data recorded in table `roundcube.userlogins`.
What is missing?

THX

**Comment by dsoares** [comment]
Hi,
Sorry for the late answer.
Please tell me the version of this plugin you have installed and show me the output of `show create table userlogins`

**Comment by 74cmonty** [comment]
Hi,

in composer.json I have defined this plugin:
"dsoares/lastlogin": "0.1.4"

Output of show create table userlogins:
```
MariaDB [roundcube]> show create table userlogins;
+------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table      | Create Table                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
+------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| userlogins | CREATE TABLE `userlogins` (
  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `timestamp` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  `user_id` int(10) unsigned NOT NULL DEFAULT '0',
  `username` varchar(128) CHARACTER SET utf8 COLLATE utf8_bin NOT NULL DEFAULT '',
  `sess_id` varchar(128) NOT NULL DEFAULT '',
  `ip` varchar(20) NOT NULL DEFAULT '',
  `real_ip` varchar(20) NOT NULL DEFAULT '',
  `hostname` varchar(255) NOT NULL DEFAULT '',
  `geoloc` varchar(255) NOT NULL DEFAULT '',
  `tor` tinyint(1) NOT NULL DEFAULT '0',
  PRIMARY KEY (`id`),
  KEY `user_id` (`user_id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 |
+------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)


```

**Comment by dsoares** [comment]
OK, it seems the table structure is OK. 
Do you have `lastlogin_geolocation` or `lastlogin_dns` enabled in config? If so, try disabling one at a time to check if the plugin logs to database without that options (and if so, one of them would be the culprit). Also check for any PHP error in logs.

**Comment by 74cmonty** [comment]
hm... I have neither `lastlogin_geolocation` nor `lastlogin_dns` enabled.
This is configured in /config/config.inc.php:
```
$config['plugins'] = array(
        'archive',
        'zipdownload',
        'acl',
        'managesieve',
        'attachment_reminder',
        'new_user_dialog',
        'twofactor_gauthenticator',
        'geolocation',
        'lastlogin',
        'removeattachments',
        'html5_notifier',
        'show_pgp_mime',
        'chbox',
        'automatic_addressbook',
        'rcguard',
);
```

**Comment by dsoares** [comment]
I meant `lastlogin_geolocation` and `lastlogin_dns` options in `lastlogin/config.inc.php`. If you have geolocation enabled, what method do you use (check `geolocation_fetch_method` option in `geolocation/config.inc.php`). And do you have any indication of a PHP error in logs?

**Comment by 74cmonty** [comment]
Hm... this is the content of /var/www/mail/roundcube/plugins/lastlogin/config.inc.php
```
<?php
/**
 * Lastlogin configuration file.
 */

// Default number of seconds to show the lastlogin info at login.
$config['lastlogin_timeout'] = 5;

// Number of entries to show in user login history.
$config['lastlogin_lastrecords'] = 5;

// Use geolocation plugin (boolean).
$config['lastlogin_geolocation'] = true;

// Resolve names with DNS (boolean).
$config['lastlogin_dns'] = true;

```

After disabling `lastlogin_dns` this has no effect on data recording.
`$config['lastlogin_dns'] = false;`

Which logs should I check for PHP error?
Could your plugin be impacted by plugin "twofactor_gauthenticator" that is working like charm?

**Comment by dsoares** [comment]
> Could your plugin be impacted by plugin "twofactor_gauthenticator" that is working like charm?

I don't think so.

Do you have the PHP GeoIP extension installed? (you may check with `php -i | grep -i geoip` in the command line or any other way you prefer).

> Which logs should I check for PHP error?

It depends on your system configuration. Check the roundcube logs or the main PHP error log file. Without some kind of error message, i'm just trying to guess it here... I can't reproduce your error, it is working for me.

**Comment by 74cmonty** [comment]
PHP GeoIP is installed.
```
root@vm104-mail:/# php -i | grep -i geoip
/etc/php5/cli/conf.d/20-geoip.ini,
geoip
geoip support => enabled
geoip extension version => 1.1.0
geoip library version => 1006002
geoip.custom_directory => no value => no value

```

I fully understand that you need some logs in order to identify the root cause.
However, I'm not sure which logs would be useful.
In the error log of the webserver Nginx there's no entry.
In access log of webserver Nginx I can see this:
```
10.0.0.2 - - [26/Jul/2016:11:39:42 +0200] "GET /roundcube/plugins/twofactor_gauthenticator/twofactor_gauthenticator_form.js?s=1468748662 HTTP/1.1" 200 1337 "https://mehl.<mydomain>.de/roundcube/?_task=login" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:36 +0200] "POST /roundcube/ HTTP/1.1" 302 5 "https://mehl.<mydomain>.de/roundcube/?_task=login" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:36 +0200] "GET /roundcube/?_task=mail HTTP/1.1" 200 12828 "https://mehl.<mydomain>.de/roundcube/?_task=login" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:36 +0200] "GET /roundcube/plugins/managesieve/skins/larry/managesieve_mail.min.css?s=1468691195 HTTP/1.1" 200 629 "https://mehl.<mydomain>.de/roundcube/?_task=mail" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:36 +0200] "GET /roundcube/skins/larry/mail.min.css?s=1468691195 HTTP/1.1" 200 22953 "https://mehl.<mydomain>.de/roundcube/?_task=mail" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:36 +0200] "GET /roundcube/plugins/chbox/skins/larry/chbox.css?s=1427110272 HTTP/1.1" 200 311 "https://mehl.<mydomain>.de/roundcube/?_task=mail" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:36 +0200] "GET /roundcube/program/js/treelist.min.js?s=1468691195 HTTP/1.1" 200 13213 "https://mehl.<mydomain>.de/roundcube/?_task=mail" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:36 +0200] "GET /roundcube/program/js/list.min.js?s=1468691195 HTTP/1.1" 200 26298 "https://mehl.<mydomain>.de/roundcube/?_task=mail" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:36 +0200] "GET /roundcube/plugins/archive/archive.min.js?s=1468691195 HTTP/1.1" 200 1561 "https://mehl.<mydomain>.de/roundcube/?_task=mail" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:36 +0200] "GET /roundcube/plugins/zipdownload/zipdownload.min.js?s=1468691195 HTTP/1.1" 200 2249 "https://mehl.<mydomain>.de/roundcube/?_task=mail" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:36 +0200] "GET /roundcube/plugins/jqueryui/js/i18n/jquery.ui.datepicker-de.js?s=1468691195 HTTP/1.1" 200 1001 "https://mehl.<mydomain>.de/roundcube/?_task=mail" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:36 +0200] "GET /roundcube/plugins/managesieve/managesieve.min.js?s=1468691195 HTTP/1.1" 200 21506 "https://mehl.<mydomain>.de/roundcube/?_task=mail" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:36 +0200] "GET /roundcube/plugins/removeattachments/removeattachments.js?s=1464946834 HTTP/1.1" 200 1703 "https://mehl.<mydomain>.de/roundcube/?_task=mail" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:36 +0200] "GET /roundcube/plugins/html5_notifier/html5_notifier.js?s=1458925535 HTTP/1.1" 200 4783 "https://mehl.<mydomain>.de/roundcube/?_task=mail" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:36 +0200] "GET /roundcube/plugins/chbox/chbox.js?s=1427110272 HTTP/1.1" 200 3744 "https://mehl.<mydomain>.de/roundcube/?_task=mail" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:36 +0200] "GET /roundcube/skins/larry/watermark.html HTTP/1.1" 200 207 "https://mehl.<mydomain>.de/roundcube/?_task=mail" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:37 +0200] "GET /roundcube/?_task=mail&_action=list&_refresh=1&_mbox=INBOX&_remote=1&_unlock=loading1469526036631&_=1469526036431 HTTP/1.1" 200 5384 "https://mehl.<mydomain>.de/roundcube/?_task=mail" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:37 +0200] "GET /roundcube/?_task=mail&_action=getunread&_page=1&_remote=1&_unlock=0&_=1469526036432 HTTP/1.1" 200 3194 "https://mehl.<mydomain>.de/roundcube/?_task=mail&_mbox=INBOX" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:37 +0200] "GET /roundcube/skins/larry/images/linen_header.jpg?v=514a.580 HTTP/1.1" 200 580 "https://mehl.<mydomain>.de/roundcube/?_task=mail&_mbox=INBOX" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:37 +0200] "GET /roundcube/skins/larry/images/listicons.png?v=1877.13442 HTTP/1.1" 200 13442 "https://mehl.<mydomain>.de/roundcube/?_task=mail&_mbox=INBOX" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:37 +0200] "GET /roundcube/skins/larry/images/buttons.png?v=32cf.15571 HTTP/1.1" 200 15571 "https://mehl.<mydomain>.de/roundcube/?_task=mail&_mbox=INBOX" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:37 +0200] "GET /roundcube/skins/larry/images/selector.png?v=799c.181 HTTP/1.1" 200 181 "https://mehl.<mydomain>.de/roundcube/?_task=mail&_mbox=INBOX" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:37 +0200] "GET /roundcube/skins/larry/images/splitter.png?v=2724.136 HTTP/1.1" 200 136 "https://mehl.<mydomain>.de/roundcube/?_task=mail&_mbox=INBOX" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:37 +0200] "GET /roundcube/skins/larry/images/quota.png?v=5c9e.4967 HTTP/1.1" 200 4967 "https://mehl.<mydomain>.de/roundcube/?_task=mail&_mbox=INBOX" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:37 +0200] "GET /roundcube/skins/larry/images/ajaxloader_dark.gif?v=cf1b.1849 HTTP/1.1" 200 1849 "https://mehl.<mydomain>.de/roundcube/?_task=mail&_mbox=INBOX" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:37 +0200] "GET /roundcube/program/resources/blank.tif HTTP/1.1" 200 270 "https://mehl.<mydomain>.de/roundcube/?_task=mail&_mbox=INBOX" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:37 +0200] "GET /roundcube/skins/larry/images/watermark.jpg?v=e784.5000 HTTP/1.1" 200 5000 "https://mehl.<mydomain>.de/roundcube/skins/larry/watermark.html" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:48 +0200] "GET /roundcube/?_task=settings HTTP/1.1" 200 4384 "https://mehl.<mydomain>.de/roundcube/?_task=mail&_mbox=INBOX" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:48 +0200] "GET /roundcube/skins/larry/settings.min.css?s=1468691195 HTTP/1.1" 200 5110 "https://mehl.<mydomain>.de/roundcube/?_task=settings" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:48 +0200] "GET /roundcube/plugins/managesieve/skins/larry/managesieve.min.css?s=1468691195 HTTP/1.1" 200 4863 "https://mehl.<mydomain>.de/roundcube/?_task=settings" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:48 +0200] "GET /roundcube/plugins/lastlogin/skins/larry/lastlogin.css?s=1466423567 HTTP/1.1" 200 346 "https://mehl.<mydomain>.de/roundcube/?_task=settings" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:48 +0200] "GET /roundcube/plugins/managesieve/skins/larry/images/vacation_icons.png?v=e738.767 HTTP/1.1" 200 767 "https://mehl.<mydomain>.de/roundcube/?_task=settings" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - - [26/Jul/2016:11:40:51 +0200] "GET /roundcube/?_task=settings&_action=edit-prefs&_section=lastlogin_preferences&_framed=1 HTTP/1.1" 200 3577 "https://mehl.<mydomain>.de/roundcube/?_task=settings" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
10.0.0.2 - thomas@<mydomain>.de [26/Jul/2016:11:42:46 +0200] "PROPFIND / HTTP/1.1" 301 178 "-" "DAVdroid/1.2 (2016/07/03; dav4android; okhttp3) Android/5.1.1"
10.0.0.2 - thomas@<mydomain>.de [26/Jul/2016:11:43:17 +0200] "PROPFIND / HTTP/1.1" 301 178 "-" "DAVdroid/1.2 (2016/07/03; dav4android; okhttp3) Android/5.1.1"

```

How can I enable further logging?

**Comment by dsoares** [comment]
It won't be necessary, i think i've found the cause.
Check the option `log_logins` in your roundcube configuration:

```
// Log successful/failed logins to <log_dir>/userlogins or to syslog
$config['log_logins'] = true;
```

If it is disabled, lastlogins won't log the user login. I should put that in the README / instructions. I'll wait for your feedback to close this issue.

**Comment by 74cmonty** [comment]
Congrats... you identified the root cause.
After enabling `$config['log_logins'] = true;` in .../roundcube/config/defaults.inc.php this issue is solved.

However, there is another issue with content of column "location". Why is there two commas inserted?
![2016-07-26_13-39-07](https://cloud.githubusercontent.com/assets/12523591/17136592/9b68d21c-5336-11e6-8232-d771104b2f0f.png)

And I would recommend to align the column headers left and include column separator line to get a better overview.
Maybe you can guide me to the php-coding where the table layout is defined and I can play around to change it.

**Comment by 74cmonty** [comment]
Shall I open another issue?
Would make sense to keep track of the different issues...

**Comment by dsoares** [comment]
Yes, it makes sense, thank you.

**Comment by dsoares** [comment]
> And I would recommend to align the column headers left and include column separator line to get a better overview.

Implemented with c10b128f66, included in release [0.1.6](https://github.com/dsoares/Roundcube-Plugin-Lastlogin/releases/tag/0.1.6).

**Comment by 74cmonty** [comment]
great stuff... like the new table design!
